tests/integration/test_lambda.py

```diff
@@ -1937,6 +1938,87 @@ def test_destroy_idle_containers(self):
         testutil.delete_lambda_function(func_name)
 
 
+def test_kinesis_lambda_parallelism(lambda_client, kinesis_client):
+    old_config = config.SYNCHRONOUS_KINESIS_EVENTS
+    config.SYNCHRONOUS_KINESIS_EVENTS = False
+    function_name = "lambda_func-{}".format(short_uid())
+    stream_name = "test-foobar-{}".format(short_uid())
+
+    testutil.create_lambda_function(
+        handler_file=TEST_LAMBDA_PARALLEL_FILE,
+        func_name=function_name,
+        runtime=LAMBDA_RUNTIME_PYTHON36,
+    )
+
+    arn = aws_stack.kinesis_stream_arn(stream_name, account_id="000000000000")
+
+    lambda_client.create_event_source_mapping(EventSourceArn=arn, FunctionName=function_name)
+
+    def process_records(record):
+        print("Processing {}".format(record))
+
+    aws_stack.create_kinesis_stream(stream_name, delete=True)
+    kinesis_connector.listen_to_kinesis(
+        stream_name=stream_name,
+        listener_func=process_records,
+        wait_until_started=True,
+    )
+
+    kinesis = aws_stack.connect_to_service("kinesis")
+    stream_summary = kinesis.describe_stream_summary(StreamName=stream_name)
+    assert 1 == stream_summary["StreamDescriptionSummary"]["OpenShardCount"]
+    num_events_kinesis = 10
+    # assure async call
+    start = time.perf_counter()
+    kinesis.put_records(
+        Records=[
+            {"Data": '{"batch": 0}', "PartitionKey": "test_%s" % i}
+            for i in range(0, num_events_kinesis)
+        ],
+        StreamName=stream_name,
+    )
+    assert (time.perf_counter() - start) < 1  # this should not take more than a second
+    kinesis.put_records(
+        Records=[
+            {"Data": '{"batch": 1}', "PartitionKey": "test_%s" % i}
+            for i in range(0, num_events_kinesis)
+        ],
+        StreamName=stream_name,
+    )
+    for _ in range(1, 10):
```

Comment:
nitpick: Could be a potential candidate for using the `retry(..)` function from `utils/common.py`
```
def get_logs():
    events = get_lambda_log_events(function_name)
    assert len(events) == 2
    return events
events = retry(get_logs, retries=5, sleep=1)
```